### ISSUE TITLE
Reader: No accessible name for Share, Comment, and Like (star) buttons

### ISSUE THREAD

**Issue by jeryj** [Automattic/wp-calypso#46737]
The buttons/links at the bottom of each reader post could use some accessibility improvements to make the work better with Screen Readers.

- [x] Share (added in #48902)
- [ ] Comment
- [ ] Like

#### Steps to reproduce

1. Starting at URL: https://wordpress.com/read
2. Use a Screen Reader (SR) to access the Share button
3. SR announces "Button" without any accessible name
4. Go to the Star/Like button
5. SR announces "Image"
6. Focus the number next to the star. Announces the number but no context as to what it is for

#### What I expected
- Share button: SR would announce "Share, button," "Share post, button," or "Share [post title], button"
- Comments: This should be a `<a>` link to the comments instead of a `<div>` along with the post title so it can be differentiated from other comment links, such as "[number] Comment(s): [post tile]"
- Star/Like: This would be a `<button aria-pressed="false">` that contains the number as well, so it would announce, "[number] Likes, toggle button" when unpressed and, "[number] Likes, selected, toggle button" when pressed.

#### Screenshot / Video
![reader screen reader issue](https://user-images.githubusercontent.com/967608/97033215-9d7ca980-1528-11eb-9d9c-bba001591c83.gif)

**Comment by synora** [comment]
@bluefuton I noticed you mentioned this issue in #48902, can this one be closed?

**Comment by bluefuton** [comment]
Hi @synora! This is part-resolved:

* https://github.com/Automattic/wp-calypso/pull/48902 added an accessible name for Reader Share
* https://github.com/Automattic/wp-calypso/pull/49691 aimed to add an accessible name for Comments but has not been completed/merged
* The Like button still does not have an accessible name

I've added a checklist to the issue description above.
